src/transformers/models/gpt_neox/modeling_gpt_neox.py

```diff
@@ -314,21 +315,39 @@ def forward(
         layer_past=None,
         output_attentions=False,
     ):
-        residual = hidden_states
-        ln_out = self.input_layernorm(hidden_states)
-        attention_layer_outputs = self.attention(
-            ln_out,
-            attention_mask=attention_mask,
-            layer_past=layer_past,
-            head_mask=head_mask,
-            use_cache=use_cache,
-            output_attentions=output_attentions,
-        )
-        attn_output = attention_layer_outputs[0]  # output_attn: a, present, (attentions)
-        outputs = attention_layer_outputs[1:]
+        if self.gpt_j_residual:
+            residual = hidden_states
+            ln_out = self.input_layernorm(hidden_states)
+            attention_layer_outputs = self.attention(
+                ln_out,
+                attention_mask=attention_mask,
+                layer_past=layer_past,
+                head_mask=head_mask,
+                use_cache=use_cache,
+                output_attentions=output_attentions,
+            )
+            attn_output = attention_layer_outputs[0]  # output_attn: a, present, (attentions)
```

Comment:
Maybe leave everything until this line outside of the if block? Duplicating this code doesn't serve any purpose here.

Reply:
Thanks for your review! I fixed it in 70aaec5c72d4613b7edb000e2d217a45d9ee78ca